src/drift.core/model/Pattern.cs

```diff
@@ -1,7 +1,11 @@
+using System;
+
 namespace Rangers.Antidrift.Drift.Core
 {
-    public class Pattern
+    public abstract class Pattern
     {
         public string Name { get; set; }
+
+        public abstract string GenerateDriftReport(TeamProject teamProject);
```

Comment:
How about making this virtual async? The idea is to impose, in case the method is overridden, to be implemented as async. If there are better ways to do that, I'll be happy to learn it.

Reply:
As pattern is still abstract, I want to stay a bit strict, until their is a reason to make it virtual. Changed the code a bit, so it returns a Task.

Reply:
Ended up making it virtual, so I can mock the method(s)